src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+
+    private int size;
+
+    private void isIndexOk(int index) {
+        if (index < 0 || index + 1 > size) {
+            throw new IndexOutOfBoundsException("Index can`t be < 0 and > size, index = " + index
+                    + " size = " + size);
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> oldNode = first;
+        for (int i = 0; i < index; i++) {
+            oldNode = oldNode.next;
+        }
+        return oldNode;
+    }
+
+    private void unLink(Node<T> node) {
+        Node<T> prev = node.previous;
+        Node<T> next = node.next;
+        if (size == 1) {
+            first = null;
+            last = null;
+        }
+        if (next == null && prev == null) {
+            size = 1;
+            return;
+        }
+
+        if (next == null) {
+            prev.next = null;
+            last = prev;
+        } else {
+
+            if (prev == null) {
+                next.previous = null;
+                first = next;
+            } else {
+                prev.next = next;
+                next.previous = prev;
+            }
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (first == null) {
+            newNode.next = null;
+            newNode.previous = null;
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element, int index) {
+        size++;
+        isIndexOk(index);
+        size--;
+        Node<T> newNode = new Node(element);
+        if (index == size) {
+            add(element);
+        } else if (index == 0) {
+            Node<T> oldNode = first;
+            oldNode.previous = newNode;
+            newNode.next = oldNode;
+            first = newNode;
+            size++;
+        } else {
```

Comment:
Playing with size in such a way is counterintuitive and may cause confusion.
It's better to change order of blocks instead:
```suggestion
        if (index == size) {
            add(element);
            return;
        }
        isIndexOk(index);
        Node<T> newNode = new Node(element);
        if (index == 0) {
            Node<T> oldNode = first;
            oldNode.previous = newNode;
            newNode.next = oldNode;
            first = newNode;
            size++;
        } else {
```